src/molgenis/capice/main_train.py

```diff
@@ -155,31 +180,26 @@ def train(self, test_set: pd.DataFrame, train_set: pd.DataFrame):
             random_state=self.model_random_state,
             use_label_encoder=False
         )
-        ransearch1 = RandomizedSearchCV(estimator=model_estimator,
-                                        param_distributions=param_dist,
-                                        scoring='roc_auc', n_jobs=8,
-                                        cv=self.cross_validate,
-                                        n_iter=self.n_iterations,
-                                        verbose=verbosity)
-
-        if int(xgb.__version__.split('.')[0]) > 0:
-            eval_set = [(test_set[self.processed_features],
-                         test_set[TrainEnums.binarized_label.value])]
-        else:
-            eval_set = [(test_set[self.processed_features],
-                         test_set[TrainEnums.binarized_label.value],
-                         'test')]
+        randomised_search_cv = RandomizedSearchCV(estimator=model_estimator,
+                                                  param_distributions=param_dist,
+                                                  scoring='roc_auc', n_jobs=8,
+                                                  cv=self.cross_validate,
+                                                  n_iter=self.n_iterations,
+                                                  verbose=verbosity)
+
+        eval_set = self._create_eval_set(xgb.__version__, test_set)
+
         self.log.info('Random search starting, please hold.')
-        ransearch1.fit(train_set[self.processed_features],
-                       train_set[TrainEnums.binarized_label.value],
-                       early_stopping_rounds=self.esr,
-                       eval_metric=["auc"],
-                       eval_set=eval_set,
-                       verbose=xgb_verbosity,
-                       sample_weight=train_set[TrainEnums.sample_weight.value])
+        randomised_search_cv.fit(train_set[self.processed_features],
+                                 train_set[TrainEnums.binarized_label.value],
+                                 early_stopping_rounds=self.esr,
+                                 eval_metric=["auc"],
+                                 eval_set=eval_set,
+                                 verbose=xgb_verbosity,
+                                 sample_weight=train_set[TrainEnums.sample_weight.value])
         self.log.info(
             'Training successful, '
             'average CV AUC of best performing model: %.4f',
-            ransearch1.best_score_)
+            randomised_search_cv.best_score_)
```

Comment:
Just using `best_score_` might be a bit too vague here... I would also like a log message with the best parameters `randomised_search_cv.best_params_`